check_certs.ps1

```diff
@@ -0,0 +1,74 @@
+##########################################################################################
+#
+# Name:        check_certs.ps1
+# Date:        2023-11-15
+# Author:      stefan.petronijevic@atomia.com
+# Version:     1.1.0
+# Parameters:  Without parameters, default values 15 and 30 are used for warning and
+#              critical, respectively. Custom values can be used.
+#              
+#              .\check_certs.ps1 10 20
+#                  10 - value for WARNING
+#                  20 - value for CRITICAL
+#
+# Returns:
+#              0 - OK
+#              1 - WARNING
+#              2 - CRITICAL
+#
+# Description: Nagios plugin that alerts whenever some certificate from store is due for
+#              renewal, and alerts based on the number of days until the expiration.
+#
+##########################################################################################
+
+param(
+    [int]$CriticalThreshold = 15, # Default value if not specified
+    [int]$WarningThreshold = 30   # Default value if not specified
+)
+
+# Initialize the exit code - 0 means no certificates are near expiration
+$exitCode = 0
+
+# Check if warning threshold is less than critical threshold and swap if necessary
+if ($WarningThreshold -lt $CriticalThreshold) {
+    $temp = $WarningThreshold
+    $WarningThreshold = $CriticalThreshold
+    $CriticalThreshold = $temp
+}
+
+# Specify the certificate store to check
+$storeLocation = "Cert:\LocalMachine\My"
+
+# Get all certificates from the specified store
+$certificates = Get-ChildItem -Path $storeLocation
+
+# Current date for comparison
+$currentDate = Get-Date
+
+foreach ($cert in $certificates) {
+    # Get certificate expiration date
+    $expirationDate = $cert.NotAfter
+
+    # Calculate days until expiration
+    $daysUntilExpiration = ($expirationDate - $currentDate).Days
+
+    # Check if the certificate is close to expiration
+    if ($daysUntilExpiration -le $CriticalThreshold) {
+        Write-Host "CRITICAL: Certificate $($cert.Subject) will expire in $daysUntilExpiration days!"

```

Comment:
If the certificate has already expired the value of `$daysUntilExpiration` will be a negative value, which may be confusing. Please reformat the message to a format that indicates that the certificate has expired it would be good if the message contained a date of expiration.

Reply:
Output now states if the certificate has already expired, along with the expiration date. Other messages are also reformated to output both days until expiration and the exact expiration date.